opa/opa.go

```diff
@@ -96,8 +98,30 @@ func (opa *client) InitializePolicy(policy string) ClientError {
 	}
 
 	if !exists {
-		// TODO: fetch rules from ES
-		return NewClientError("policy does not exist", OpaClientErrorTypePolicyNotFound, nil)
+		err := opa.loadPolicy(policy, policyData)
+		if err != nil {
+			return NewClientError(err.Error(), OpaClientErrorTypeLoadPolicy, err)
+		}
+	}
+
+	return nil
+}
+
+func (opa *client) loadPolicy(policy string, policyData string) error {
+	log := opa.logger.Named("Load Policy")
+	client := &http.Client{}
```

Comment:
Most likely want to set a timeout here. IIRC the default client doesn't have one. It probably also makes sense to re-use the client instead of instantiating it on every call.

Reply:
@alexashley in other places we are just using ` http.DefaultClient.Do(req)` . You're saying that it doesn't have a default timeout. Should i instead be creating an http client in here 
```

Reply:
Yeah I think it makes sense to create it either in the constructor or as an argument to the constructor and then reuse it. 
Could definitely be part of the follow up too, doesn't have to be in this PR.

Reply:
Cool. this is a good idea. Handling this in the cleanup PR which will include
Sorting, Esutils package, and this (http client argument with a timeout)